core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedInPredicateToJoin.java

```diff
@@ -81,7 +81,7 @@
  * Into:
  * <pre>
  * - Project (output: CASE WHEN (countmatches > 0) THEN true WHEN (countnullmatches > 0) THEN null ELSE false END)
- *   - Aggregate (countmatches=count(*) where a, b not null; countnullmatches where a,b null but buildSideKnownNonNull is not null)
+ *   - Aggregate (countmatches=count(*) where a, b not null; countnullmatches where (a is null or b is null) but buildSideKnownNonNull is not null)
  *     grouping by (A'.*)
  *     - LeftJoin on (A and B correlation condition)
```

Comment:
Could you please also update this line? The join condition, besides the correlated predicate,  also includes the "in" condition (and additional predicates to pass the null values from either side of the join).